weis/glue_code/runWEIS.py

```diff
@@ -32,81 +33,32 @@ def run_weis(fname_wt_input, fname_modeling_options, fname_opt_options, geometry
     myopt = PoseOptimizationWEIS(wt_init, modeling_options, opt_options)
-
+    
     if MPI:
-        n_DV = myopt.get_number_design_variables()
-        # Extract the number of cores available
-        max_cores = MPI.COMM_WORLD.Get_size()
-
-        # Define the color map for the parallelization, determining the maximum number of parallel finite difference (FD)
-        # evaluations based on the number of design variables (DV). OpenFAST on/off changes things.
-        if modeling_options['Level3']['flag']:
-
-            # If we are running an optimization method that doesn't use finite differencing, set the number of DVs to 1
-            if not (opt_options['driver']['design_of_experiments']['flag']) and (opt_options['driver']['optimization']['solver'] in evolutionary_methods):
-                n_DV *= 5  # targeting 10*n_DV population size... this is what the equivalent FD coloring would take
-            elif not (opt_options['driver']['design_of_experiments']['flag'] or opt_options['driver']['optimization']['solver'] in fd_methods):
-                n_DV = 1
-
-
-            # If openfast is called, the maximum number of FD is the number of DV, if we have the number of cores available that doubles the number of DVs,
-            # otherwise it is half of the number of DV (rounded to the lower integer).
-            # We need this because a top layer of cores calls a bottom set of cores where OpenFAST runs.
-            if max_cores > 2. * n_DV:
-                n_FD = n_DV
-            else:
-                n_FD = int(np.floor(max_cores / 2))
-            # Get the number of OpenFAST runs from the user input and the max that can run in parallel given the resources
-            # The number of OpenFAST runs is the minimum between the actual number of requested OpenFAST simulations, and
-            # the number of cores available (minus the number of DV, which sit and wait for OF to complete)
-            n_OF_runs = modeling_options['DLC_driver']['n_cases']
-            n_DV = max([n_DV, 1])
-            max_parallel_OF_runs = max([int(np.floor((max_cores - n_DV) / n_DV)), 1])
-            n_OF_runs_parallel = min([int(n_OF_runs), max_parallel_OF_runs])
-        elif modeling_options['Level2']['flag']:
-
-            if not (opt_options['driver']['design_of_experiments']['flag'] or opt_options['driver']['optimization']['solver'] in fd_methods):
-                n_DV = 1
-
-
-            if max_cores > 2. * n_DV:
-                n_FD = n_DV
-            else:
-                n_FD = int(np.floor(max_cores / 2))
-            n_OF_runs = modeling_options['Level2']['linearization']['NLinTimes']
-            n_DV = max([n_DV, 1])
-            max_parallel_OF_runs = max([int(np.floor((max_cores - n_DV) / n_DV)), 1])
-            n_OF_runs_parallel = min([int(n_OF_runs), max_parallel_OF_runs])
+        if not prepMPI:
+            nFD = modeling_options['General']['openfast_configuration']['nFD']
```

Comment:
Does the user now need to enter `nFD` and `nOFp` into the modeling options?

Reply:
I believe this comment is now outdated. the users can provide those two inputs, but it's a lot safer to let WEIS compute them with the --preMPI flag

Reply:
so the short answer is no, the user does not need to provide those two inputs